src/rgw/rgw_bucket.cc

```diff
@@ -558,6 +561,39 @@ int rgw_remove_bucket(RGWRados *store, rgw_bucket& bucket, bool delete_children)
 
   }
 
+  do {
+
+    objs.clear();
+    string prefix = "", marker = "", delimiter = "";
+    static MultipartMetaFilter mp_filter;
+    bool is_truncated;
+
+    ret = list_bucket_multiparts(store, info, prefix, marker, delimiter, &mp_filter,
+				max, &objs, &common_prefixes, &is_truncated);
+
+    if (ret < 0) {
+      return ret;
+    }
+
+    if (!objs.empty() && delete_children) {
+      vector<rgw_bucket_dir_entry>::iterator iter;
+      RGWMultipartUploadEntry entry;
+      for (iter = objs.begin(); iter != objs.end(); ++iter) {
+        rgw_obj_key key(iter->key);
+        if (!entry.mp.from_meta(key.name))
+          continue;
+        entry.obj = *iter;
+        abort_multipart_upload(store,cct,&obj_ctx,info,entry.mp);
+        ldout(store->ctx(),0) << "WARNING : cleaned incomplete multipart" << dendl;
```

Comment:
this could get spammy, maybe just count these up and print a single warning after the loops? i.e. `WARNING: aborted 15 incomplete multipart uploads`